app/logic/Calendar/Event.ts

```diff
@@ -153,12 +153,12 @@ export class Event extends Observable {
    */
   private setRecurrenceRule(rule: RecurrenceRule) {
     assert(this.recurrenceCase == RecurrenceCase.Normal || this.recurrenceCase == RecurrenceCase.Master, "Instances can't themselves recur");
+    this._recurrenceRule = rule;
+    this.recurrenceCase = RecurrenceCase.Master; // notifies
     if (!this._recurrenceRule?.timesMatch(rule)) {
       this.clearExceptions();
     }
-    this._recurrenceRule = rule;
-    this.recurrenceCase = RecurrenceCase.Master; // notifies
-    this.instances.clear();
+    this.generateRecurringInstances();
```

Comment:
Oops, you're comparing the rule after you replace it...

Reply:
Fixed by Neil in #671